File: tools/check_supported_home_assistant.py
#!/usr/bin/env python3
"""Validate the supported Home Assistant version across repository metadata."""

from __future__ import annotations

import json
import re
from pathlib import Path

SUPPORTED_VERSION = "2026.7.0"

hacs = json.loads(Path("hacs.json").read_text(encoding="utf-8"))
if hacs.get("homeassistant") != SUPPORTED_VERSION:
    raise SystemExit(f"hacs.json must declare Home Assistant {SUPPORTED_VERSION}")

requirements = Path("requirements-dev.in").read_text(encoding="utf-8")
match = re.search(r"^homeassistant==([^\s]+)$", requirements, re.MULTILINE)
if match is None or match.group(1) != SUPPORTED_VERSION:
    raise SystemExit(
        f"requirements-dev.in must pin Home Assistant {SUPPORTED_VERSION}"
    )

documentation = Path("docs/compatibility.md").read_text(encoding="utf-8")
if f"Home Assistant {SUPPORTED_VERSION} or newer" not in documentation:
    raise SystemExit("Compatibility documentation is out of sync")

print(f"Supported Home Assistant version is consistent: {SUPPORTED_VERSION}")

File: scripts/check-supported-home-assistant.mjs
import { readFile } from "node:fs/promises";

const supported = "2026.7.0";
const hacs = JSON.parse(await readFile("hacs.json", "utf8"));
const requirements = await readFile("requirements-dev.in", "utf8");
const documentation = await readFile("docs/compatibility.md", "utf8");

if (hacs.homeassistant !== supported) {
  throw new Error(`hacs.json must declare Home Assistant ${supported}`);
}
if (!requirements.includes(`homeassistant==${supported}`)) {
  throw new Error(`requirements-dev.in must pin Home Assistant ${supported}`);
}
if (!documentation.includes(`Home Assistant ${supported} or newer`)) {
  throw new Error("Compatibility documentation is out of sync");
}

console.log(`Supported Home Assistant version is consistent: ${supported}`);
